### ISSUE TITLE
Documentation for the firestore reference data type required

### ISSUE THREAD

**Issue by RobinManoli** [flutter/flutter#24819]
Personally I would like to be able to decide whether referenced documents' data is included in queries or not. It seems to me that referenced documents each need a second request to be retrieved. However, it would of course be nice to know if this were true.

Not only in flutter but in other frameworks (such as VueFire) the reference data type seems to be inconsistently handled, which means it is not reliable to use the reference data type.

At first glance I was very happy that firestore has a reference field, but in practice it took a lot more time to understand how to use it poorly than to just use strings and retrieve the documents manually.

I think it's certainly an issue to implement the reference type (without a warning) if they make the development process more difficult than implementing the functionality yourself.

**Comment by zoechi** [comment]
Is this a Firestore issue or a Flutter Firestore plugin issue?
Any link to how it's done in Android native and what is inferior in Flutter compared to that?

**Comment by RobinManoli** [comment]
The lack of documentation is a Flutter Firestore plugin issue.

Then I don't know how much of the issue is about the flutter implementation compared to how firestore actually works. The reference data type might work well on some other platform, but of course the official firebase documentation could explain things better.

I can't compare the issue with Android native as I'm not an Android native developer.

**Comment by zoechi** [comment]
You could try a StackOverflow question to learn more and then update the issue when it's more clear what is missing. 
(I haven't used Firestore and don't understand what this issue is about).

**Comment by RobinManoli** [comment]
Sorry if I wasn't clear enough. Let me try again:

1. There is no documentation for how to use the reference data type of firestore, I would like to kindly ask any developer to provide this
2. I would like the documentation to be clear on how to intelligibly use the data type, and be explicit on whether you need to send an extra request to receive the data for each reference data type
3. Even if it does work intelligibly to use this datatype on flutter, I also would like to issue a warning about using it, as it was not only for flutter that it was poorly documented, and any project successfully using the reference data type in flutter might stumble on to problems in other frameworks after having implemeneted this datatype in flutter - forcing them to revert their flutter project. (I understand this point might not count as a flutter issue, but I would really like to warn the internet before using this firestore feature, just like I would like to have been warned before I attempted to use it)

**Comment by shadimac** [comment]
@RobinManoli for me I have solved it the old way, I didn't use `reference` data type even, I have used a normal `String` data to store the other document Id, and during implementation, I retrieve the other document from the collection using the document Id.

               ```
return FutureBuilder(
    future: _getAnotherDocument(documentId),
    builder: (context, snapshot) {
        DocumentSnapshot otherDoc = snapshot.data; 
        return otherDoc["fieldName"]; //here you have access to the other document
     }
);
```

And inside the _getAnotherDocument you can access the corresponding collection to get back the document using the document Id:

```
Future<DocumentSnapshot> _getAnotherDocument(String documentId) async {
    return await Firestore.instance.collection("collectionName").document(documentId).get().then((docSnap) {
              return docSnap;
        }
    );
}
```

And [here](https://stackoverflow.com/questions/46568850/what-is-firestore-reference-data-type-good-for) you can find another solution similar to this one but using the reference data type.

**Comment by kroikie** [comment]
@RobinManoli

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/565. Any further collaboration will be done there.
